### DISCUSSION TITLE
"Import()" has been deprecated, use "import x" instead

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#16083] [General]
What's the deal with deprecating the round brackets for` import` `()`?  Doesn't seem consistent with `const`, `global`, etc... having the ability to use round brackets for a list, but not `import`.

**Comment by spytheman** [comment]
Using `import name`, makes searching for all imports in a module extremely simple. The grouping () lists make searching harder for no good reason.

**Reply by Wajinn** [reply]
Hmmm... I understand your argument, but it does look inconsistent.

**Reply by spytheman** [reply]
It is - imho `const` and global declarations should *not* support grouping as well, for the same reason.

**Reply by Wajinn** [reply]
Well, I guess its one of the rare times I would disagree.  Just saying from a user's perspective and as an opinion, the groupings increased usability and is/was nice.  Not anything major to lose sleep over, but liked the feature of having the groupings.

**Reply by JalonSolov** [reply]
It is visually less cluttered.  It is much harder to search.

**Reply by Wajinn** [reply]
I kind of understand the developer perspective of making it easier to search, but then it was decided to add the feature in the first place, so would of thought that was worked out a while ago.  I'm assuming the original design decision was for user convenience, so now it looks like the case of taking a feature away.  It looks, at least, a bit contradictory.  I would hope the trend would stop at just `import` or even revert back to how it was.

When it comes to being less visually cluttered, sorry, just not seeing that.  Its almost like a violation of DRY, to keep retyping the same keyword multiple times.  Of course, I acknowledge that it's a matter of preference.

```
import w
import x
import y
import z

__global w = 1
__global x = 2
__global y = 3
__global z = 4

const w = 1
const x = 2
const y = 3
const z = 4
```
```
import (
w
x
y
z
)

__global (
w = 1
x = 2
y = 3
z = 4
)

const (
w = 1
x = 2
y = 3
z = 4
)
```

**Reply by JalonSolov** [reply]
I meant it is less visually cluttered to use `import ()`, but it is harder to search for _just_ imports.

**Reply by Wajinn** [reply]
> I meant it is less visually cluttered to use `import ()`, but it is harder to search for _just_ imports.

Oh, understand now.  I'm of course sympathetic to the developers perspective, just giving a user's view of how it might come across.

**Reply by bogen85** [reply]
Visual sugar candy is not always conducive to maintaining good reliable code, as it can bog down the maintenance efforts.

I agree it is a tradeoff, but at the end of the day, when it comes to something being either beginner and casual reader friendly, or developer friendly, I'll take the latter.

If it can be both, of course I'll take that option.

In what has been highlighted above, WET (write everything twice) vs DRY (don't repeat yourself)

Although the `T` is more than twice...

I agree the DRY looks better and is more visually appealing.

With the proper cross referencing tools it could be developer friendly. But that complicates the tool-set developers need to navigate, understand, and maintain code.

**Comment by changrui** [comment]
`import (a, b, c)`.

**Reply by bogen85** [reply]
Force it all one the same line?

**Reply by bogen85** [reply]
But that is not a requirement for other `(...)`

**Reply by bogen85** [reply]
I guess the `v-fmt` could break it up into as many lines as needed? but what if the programmer does not want that?

**Reply by JalonSolov** [reply]
`import (a b c)` (without the commas) is what we used to have.  But `v fmt` would change it to
```v
import (
    a
    b
    c
)
```
